### ISSUE TITLE
Processing of messages from the socket should split the json

### ISSUE THREAD

**Issue by BenoitMercier** [pradeepmouli/udi-js#62]
If 2 messages are sent quickly, the service will read the 2 messages as a single string and fail to parse the json.

Example:
Error processing msg: {"type": "command", "command": "start"}{"type": "matterConfig"}: Unexpected token { in JSON at position 39

**Comment by pradeepmouli** [comment]
I think I do handle the splitting the JSON on the server side.... already....:

```
 .on('data', async (data: any) => {
				logger.debug(`Received: ${data.toString()}`);
				let s = data.toString().trim().split('\n');
				for (let line of s) {
					await processMessage(line);
				}
				return;
			})
```

**Comment by BenoitMercier** [comment]
No. If the plugin sends 2 messages one after the other, only a single data event will be fired. In the example above, I was sending 2 messages, a start command, following a matterConfig. The matter-server got a single data event and choked with what was 2 JSON in a single string.

I encountered the same problem in the plugin. On connecting, I could receive up to 3 JSON in a single read event, so I had to make code to split it into multiple JSONs and process them separately.

**Comment by pradeepmouli** [comment]
right... i've been splitting the message _before_ parsing the JSON, so it should be okay... no?

**Comment by BenoitMercier** [comment]
That should work if the plugin manually adds a newline at the end of each message.

**Comment by pradeepmouli** [comment]
it should handle the single message case w/o a trailing new line.... at least that's how it worked in my testing

**Comment by BenoitMercier** [comment]
Perhaps in your testing you were sending formatted json which included a trailing newline?

I assure you the log I posted came from the matter server as is.

**Comment by BenoitMercier** [comment]
Actually, if the JSON is formatted, it will include newlines inside the brackets. That's problematic because a single json will be fragmented in multiple parts.

Since only the plugin is expected to talk to this server, we can use this work-around:
1. Plugin should always send compact JSONs ending with a newline
2. Matter server should continue to use split('\n'), but it should also filter for empty string. 
(JSON.parse('') will trow.)

I think that's the easiest solution (It's easier that splitting 2 json in a single string). Not super clean, but acceptable given that only the plugin should ever write to this socket.
